### PR TITLE
globalkeysd refactoring (session 1)

### PR DESCRIPTION
**Pull request by antis81**
Session 1:
- ports code to modern C++ highly improving it's readability
- moves X11 event loop handling code into a dedicated function.

### CODE REVIEW

File: daemon/core.h [hunk 1]
```diff
@@ -100,52 +100,52 @@ class Core : public QThread, public LogTarget
     using ClientPaths           = QSet<ClientPath>;
     using ClientPathsBySender   = QMap<QString, ClientPaths>;
 
-private slots:
-    void serviceDisappeared(const QString &sender);
+private:
+    Q_INVOKABLE void serviceDisappeared(const QString &sender);
```
Comment (palinek) [inline daemon/core.h:104]: Do we even need this methods to be available to Qt meta system when we're using the modern (Qt5) signal/slot connection?

Comment (antis81) [reply daemon/core.h:104]: Good question! Somehow didn't realize it is possible now to connect function pointers directly. Thanks!

**Comment by antis81** [comment]
@tsujan Because of the complexity of this topic I think we can merge at this (stable) point and continue work in a new PR. If we continue work here, changes will now get more and more difficult to review.

**Comment by tsujan** [comment]
> Because of the complexity of this topic I think we can merge at this...

Merging of a PR requires the approval of a reviewer. @palinek hasn't approved it yet and I even haven't had time to take a look at it.

**Review by palinek — APPROVED** [review]
I just looked the commit diffs changes one-by-one. All seem good to me. Simple compile & run succeeded for me.

**Comment by tsujan** [comment]
@palinek (or LXQt members),

Please merge it if you're sure about its sanity. I won't have time soon to review such a massive reorganization.

**Comment by stefonarch** [comment]
Tested compiling and daily use, no issues found.

**Pull request merged**